src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -63,28 +64,38 @@ open class KtlintPlugin : Plugin<Project> {
         }
     }
 
-    private fun addKtLintTasksToAndroidKotlinPlugin(target: Project,
-                                                    extension: KtlintExtension) {
+    private fun addKtLintTasksToAndroidKotlinPlugin(target: Project, extension: KtlintExtension) {
         target.pluginManager.withPlugin("kotlin-android") {
             target.afterEvaluate {
                 val ktLintConfig = createConfiguration(target, extension)
-                
-                target.extensions.findByType(BaseExtension::class.java)?.sourceSets?.forEach {
-                    val kotlinSourceDir = it.java.sourceFiles
-                    val runArgs = it.java.srcDirs.map { "${it.path}/**/*.kt" }.toMutableSet()
+
+                val variantManager = when {
+                    target.plugins.hasPlugin(AppPlugin::class.java) -> target.plugins.findPlugin(AppPlugin::class.java)?.variantManager
+                    target.plugins.hasPlugin(LibraryPlugin::class.java) -> target.plugins.findPlugin(LibraryPlugin::class.java)?.variantManager
+                    target.plugins.hasPlugin(InstantAppPlugin::class.java) -> target.plugins.findPlugin(LibraryPlugin::class.java)?.variantManager
```

Comment:
Hi @originx, is it wanted to use always the LibraryPlugin? Probably it will be better to use the specific Plugin, in this case InstantAppPlugin.

Reply:
yep thanks, rush of duplicating lines, fixed now :)